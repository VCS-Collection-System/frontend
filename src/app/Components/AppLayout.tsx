import * as React from 'react';
import { useHistory } from 'react-router-dom';
import {
  Alert,
  Page,
  Flex,
  FlexItem,
  PageHeader,
  SkipToContent,
  Title,
  Button,
} from '@patternfly/react-core';
import logo from '@app/Assets/Logo-RedHat-A-Color-RGB.svg';
import asciiLogo from '@app/Assets/Logo-RedHat-C-Color-Ascii.png';
import { useDarkMode } from  "@app/utils/customInputs"
import Feature from './Feature';
import GlobalStyle from '@app/utils/GlobalStyle';
import { MAINTENANCE_BANNER, MAINTENANCE_BANNER_TEXT } from "@app/utils/constants";

interface IAppLayout {
  children: React.ReactNode,
  loggedIn: boolean;
}

const AppLayout: React.FunctionComponent<IAppLayout> = ({ children, loggedIn}) => {
  const history = useHistory();
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [isMobileView, setIsMobileView] = React.useState(true); //NOSONAR
  const [themeSwitcher, setThemeSwitcher] = React.useState(false)
  const onPageResize = (props: { mobileView: boolean; windowSize: number }) => {
    setIsMobileView(props.mobileView);
  };
  // Extract this for re-use
  const clearAndRedirect = () => {
    sessionStorage.clear();
    window.location.href = "/";
  }

  // Figure out why we can't control page layout without isNavOpen of this component
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const Header = ( //NOSONAR
    <PageHeader
      isNavOpen={false}
    />
  );
  const pageId = 'primary-app-container';
  const PageSkipToContent = (
    <SkipToContent onClick={(event) => {
      event.preventDefault();
      const primaryContentContainer = document.getElementById(pageId);
      primaryContentContainer && primaryContentContainer.focus();
    }} href={`#${pageId}`}>
      Skip to Content
    </SkipToContent>
  );

  return (
    <Page
      style={{ backgroundColor: "rgb(222, 222, 222)" }}
      mainContainerId={pageId}
      onPageResize={onPageResize}
      skipToContent={PageSkipToContent}
    >
      {MAINTENANCE_BANNER && (<Alert variant="info" title={MAINTENANCE_BANNER_TEXT} style={{width: "100vw"}}></Alert>)}
      <Flex
        alignItems={{ default: "alignItemsCenter" }}
        direction={{ default: "column" }}
        style={{ marginTop: "1rem" }}
      >
        <Flex 
          fullWidth={{ default: "fullWidth" }} 
          justifyContent={{ default: "justifyContentSpaceBetween" }}>
          <FlexItem style={{paddingLeft: "var(--pf-l-flex--spacer)"}}>
            <img src={logo} onClick={() => history.push("/")} alt="RH Logo" width="120px" style={{marginBottom: "-12px"}} />
          </FlexItem>
          <FlexItem alignSelf={{ default: "alignSelfFlexEnd" }}>
            <Button
              onClick={() => history.push("/contact")}
              variant="plain"
            >
              <span style={{ color: themeSwitcher ? "rgb(180, 180, 178)" : "black" }}>HELP</span>
            </Button>
            <Button
              onClick={() => history.push("/privacy")}
              variant="plain"
            >
              <span style={{ color: themeSwitcher ? "rgb(180, 180, 178)" : "black" }}>PRIVACY</span>
            </Button>
            {loggedIn ? (
              <Button onClick={clearAndRedirect} variant="plain">
                <span style={{ color: themeSwitcher ? "rgb(180, 180, 178)" : "black" }}>LOGOUT</span>
              </Button>
            ) : "" }
          </FlexItem>
        </Flex>
        <FlexItem style={{margin: "var(--pf-global--spacer--3xl) 0 var(--pf-global--spacer--2xl) 0"}}>
          <Title headingLevel="h1" size="3xl" style={{ fontWeight: 300 }}>
            Validation Check-In System
          </Title>
        </FlexItem>
        <FlexItem style={{ width: "95%" }}>{children}</FlexItem>
        {useDarkMode() ? (
          <Feature name="customInput"> 
            <Flex 
              fullWidth={{ default: "fullWidth" }} 
              justifyContent={{ default: "justifyContentCenter" }}
            >
              <FlexItem>
                {themeSwitcher ? <GlobalStyle /> : ""}
                <img src={asciiLogo} onClick={() => setThemeSwitcher(!themeSwitcher)} alt="RH Logo" width="120px" style={{marginRight: "12px"}} />
              </FlexItem>
            </Flex>
          </Feature>
        ) : ""}
      </Flex>
    </Page>
  );
}

export { AppLayout };