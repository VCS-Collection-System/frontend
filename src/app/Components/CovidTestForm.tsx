import React, { useContext, useEffect, useState } from "react";
import { 
  Checkbox,
  Form, 
  FormGroup, 
  TextInput, 
  FormSelect, 
  FormSelectOption, 
  DatePicker, 
  FileUpload, 
  Button, 
  Flex,
  FlexItem,
  Text
} from "@patternfly/react-core";
import { useKeycloak } from "@react-keycloak/web";
import { CheckCircleIcon, OutlinedPaperPlaneIcon } from "@patternfly/react-icons";
import { Controller, useForm } from "react-hook-form";
import { useHistory } from "react-router-dom";
import Feature, { showFeature } from "@app/Components/Feature";
import { ConsentContext } from "@app/context";
import { 
  dateConvert, 
  americanDateFormat,
  covidTestDateRules,
  covidTestDateValidator,
  birthDateValidator,
  birthDateRules
} from "@app/utils/utils";
import {
  ALTERNATE_CONSENT_LABEL
} from "@app/utils/constants";

const CovidTestForm = (props) => {

  const history = useHistory();
  const { keycloak } = useKeycloak();
  const { control, handleSubmit } = useForm();
  const [isTestFileRejected, setIsTestFileRejected] = useState(false);
  const { setConsented } = useContext(ConsentContext);
  const [ isCovidTest, setIsCovidTest ] = useState(false); 
  const [ isRecovery, setIsRecovery ] = useState(false);
  const [allowPDF, setAllowPDF] = useState(false);

  let resultLabel = "";
  if(isCovidTest) {
    resultLabel += "COVID Test Result";
    if(isRecovery) {
      resultLabel += " or Certificate of Recovery";
    }
  } else {
    resultLabel += "Type of Submission";
  }
  resultLabel += ".";

  let resultHelper = "Please select the ";
  if(isCovidTest) {
    resultHelper += "COVID test result";
    if(isRecovery) {
      resultHelper += " or certificate of recovery";
    }
  } else {
    resultHelper += "type of submission";
  }
  resultHelper += ".";

  let dateHelper = "Please enter the date the ";
  if(isCovidTest) {
    dateHelper += "COVID test";
    if(isRecovery) {
      dateHelper += " or certificate of recovery";
    }
  } else {
    dateHelper += "certificate of recovery";
  }
  dateHelper += " was administered.";

  useEffect(() => {
    showFeature("covidTest", keycloak, props.employeeCountry).then(config => {
      setIsCovidTest(config);
    });

    showFeature("recovery", keycloak, props.employeeCountry).then(config => {
      setIsRecovery(config);
    });

    showFeature("allowPdf", keycloak, props.employeeCountry).then(config => {
      setAllowPDF(config);
    });
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const ImageHelperText = ({isInvalidText}) => {    
    return (
      <Text className={isInvalidText ? "helperText warningText" : "helperText"}> 
        Must be a {allowPDF ? "PDF," : ""} JPG, JPEG, or PNG file no larger than 3.9 MB 
        <br />
        Please refer to &nbsp;
        <a href="https://redhat.service-now.com/help?id=kb_article_view&sysparm_article=KB0015729" target="_blank"  rel="noreferrer">
          Image Submission FAQ
        </a> 
        &nbsp; for further guidance. 
      </Text>
    )
  }

  const getAcceptableProofs = () => {
    return ".jpg,.jpeg,.png" + (allowPDF ? ",.pdf" : "" );
  }

  return (
    <Form onSubmit={handleSubmit(props.onSubmit)}>
      {!props.isHRSubmit ? (
        <Feature name="consentCheckboxAlternate">
          <Controller
            name="consent"
            control={control}
            render={({ field, fieldState }) => (
              <FormGroup 
                fieldId="consent-checkbox" 
                isRequired
                label="Consent Agreement"
                helperTextInvalid="This field is required, please complete consent agreement."
                validated={fieldState.error ? "error" : "default"}
              >
                <Checkbox 
                  label={ALTERNATE_CONSENT_LABEL} 
                  aria-label="consent-checkbox" 
                  id="consent-checkbox" 
                  isChecked={field.value}
                  onChange={(value) => {
                    setConsented(value);
                    field.onChange(value);
                  }}
                />
              </FormGroup>
            )}
            rules={{ required: true }}
          />
        </Feature>
      ) : ""}
      <FormGroup
        fieldId="employee-full-name"
        helperText={
          props.isHRSubmit ? 
          "Make sure this matches the full name of the person you are submitting for." : 
          "Make sure this matches your full name."
        }
        label="Full Name"
      >
        <TextInput
          aria-label="employee-full-name"
          isDisabled
          type="text"
          value={props.employeeName}
        />
      </FormGroup>
      <Controller
        name="alt_first_name"
        control={control}
        defaultValue={props.isHRSubmit ? "" : (sessionStorage.getItem("alt_first_name") || "")}
        render={({ field, fieldState }) => (
          <FormGroup 
            fieldId="alt-first-name" 
            helperText={
              props.isHRSubmit ? 
              "Individual's first name as displayed on the documentation if different than above." : 
              "Your first name as displayed on the documentation if different than above."
            }
            label="Preferred First Name (if applicable)"
            validated={fieldState.error ? "error" : "default"}
          >
            <TextInput
              aria-label="alt-first-name"
              type="text"
              value={field.value}
              onChange={field.onChange}
            />
          </FormGroup>
        )}
      />
      <Controller
        name="alt_last_name"
        control={control}
        defaultValue={props.isHRSubmit ? "" : (sessionStorage.getItem("alt_last_name") || "")}
        render={({ field, fieldState }) => (
          <FormGroup 
            fieldId="alt-last-name" 
            helperText={
              props.isHRSubmit ? 
              "Individual's last name as displayed on the documentation if different than above." : 
              "Your last name as displayed on the documentation if different than above."
            }
            label="Preferred Last Name (if applicable)"
            validated={fieldState.error ? "error" : "default"}
          >
            <TextInput
              aria-label="alt-last-name"
              type="text"
              value={field.value}
              onChange={field.onChange}
            />
          </FormGroup>
        )}
      />
      <Feature name="dateOfBirth" employeeCountry={props.employeeCountry}>
        <Controller
          name="date_of_birth"
          control={control}
          defaultValue={props.isHRSubmit ? "" : (dateConvert(sessionStorage.getItem("date_of_birth") || ""))}
          render={({ field, fieldState }) => (
            <FormGroup
              helperText={
                props.isHRSubmit ?
                "Please enter the birth date as shown on the documentation." :
                "Please enter your birth date."
              }
              helperTextInvalid="This field is required, please enter a valid date."
              fieldId="date_of_birth"
              label="Date of Birth"
              isRequired
              validated={fieldState.error ? "error" : "default"}
            >
              <DatePicker
                aria-label="date_of_birth"
                onChange={field.onChange}
                dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
                dateParse={americanDateFormat}
                placeholder="MM-DD-YYYY"
                value={field.value}
                validators={[birthDateValidator]}
              />
            </FormGroup>
          )}
          rules={{ required: true, validate: birthDateRules, pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/ }}
        />
      </Feature>
      <Controller
        name="test_results"
        control={control}
        rules={{ required: true }}
        defaultValue={props.isHRSubmit ? "" : (sessionStorage.getItem("test_results") || "")}
        render={({ field, fieldState }) => (
          <FormGroup
            fieldId="test-results-input"
            helperText={resultHelper}
            isRequired
            validated={fieldState.error ? "error" : "default"}
            label={resultLabel}
            helperTextInvalid="This field is required to proceed."
          >
            <FormSelect
              aria-label="test-results-input"
              onChange={field.onChange}
              validated={fieldState.error ? "error" : "default"}
              value={field.value}
            >
              <FormSelectOption
                isPlaceholder
                isDisabled
                label="Please Select"
              />
              <Feature name="covidTest" employeeCountry={props.employeeCountry}>
                <FormSelectOption label="Negative" value="Negative" />
                <FormSelectOption label="Positive" value="Positive" />
                <FormSelectOption label="Inconclusive" value="Inconclusive" />
              </Feature>
              <Feature name="recovery" employeeCountry={props.employeeCountry}>
                <FormSelectOption label="Certificate of Recovery" value="Recovery" />
              </Feature>
            </FormSelect>
          </FormGroup>
        )}
      />
      <Controller
        control={control}
        defaultValue={props.isHRSubmit ? "" : (dateConvert(sessionStorage.getItem("test_date") || ""))}
        name="test_date"
        render={({ field, fieldState }) => (
          <FormGroup
            label="Date"
            fieldId="test-results-date"
            helperText={dateHelper}
            helperTextInvalid="This field is required to proceed."
            isRequired
            validated={fieldState.error ? "error" : "default"}
          >
            <DatePicker
              aria-label="test-results-date"
              dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
              dateParse={americanDateFormat}
              placeholder="MM-DD-YYYY"
              onChange={field.onChange}
              value={field.value}
              validators={[covidTestDateValidator]}
            />
          </FormGroup>
        )}
        rules={{ required: true, validate: covidTestDateRules, pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/  }}
      />
      {!props.isHRSubmit &&
        <Controller
          control={control}
          defaultValue={props.onLoad}
          name="test_image_file"
          render={({ field, fieldState }) => (
            <FormGroup
              fieldId="test-record-image"
              isRequired
              label="Provide a copy of supporting documentation."
              helperText={<ImageHelperText isInvalidText={false}/>}
              helperTextInvalid={<ImageHelperText isInvalidText={true}/>}
              validated={isTestFileRejected || fieldState.error ? "error" : "default"}
            >
              <FileUpload
                hideDefaultPreview
                filename={field.value ? field.value["name"] : ""}
                id="test-record-image"
                value={field.value}
                onChange={field.onChange}
                dropzoneProps={{
                  accept: getAcceptableProofs(),
                  maxSize: 3900000,
                  onDropRejected: () => {
                    setIsTestFileRejected(true);
                  },
                  onDropAccepted: () => {
                    setIsTestFileRejected(false);
                  }
                }}
                validated={isTestFileRejected ? "error" : "default"}
              />
            </FormGroup>
          )}
          rules={{ required: true }}
        />
      }
      {props.isHRSubmit &&
        <Controller
          name="alternate_email"
          defaultValue={""}
          control={control}
          render={({ field, fieldState }) => (
            <FormGroup
              id="alternative-email"
              label="Alternate Email (optional)"
              helperText="Optional email for notifications"
              helperTextInvalid="This needs to be a valid email address."
              fieldId="alternative-email"
              validated={fieldState.error ? "error" : "default"}
            >
              <TextInput
                aria-label="alternative-email"
                onChange={field.onChange}
                placeholder="example@mail.com"
                type="email"
                value={field.value}
              />
            </FormGroup>
          )}
          rules={{
            required: false,
            pattern: /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}/g, // NOSONAR
          }}
        />
      }
      {props.isHRSubmit ? 
        <FormGroup fieldId="accept-test-buttons">
          <Flex>
            <FlexItem align={{ default: "alignRight" }}>
              <Button
                icon={<OutlinedPaperPlaneIcon />}
                variant="primary"
                type="submit"
                isDisabled={props.isFormSubmitting}
                isLoading={props.isFormSubmitting}
              >
                Submit
              </Button>
            </FlexItem>
          </Flex>
        </FormGroup>
        :
        <FormGroup fieldId="accept-test-buttons">
          <Button 
            icon={<CheckCircleIcon />} 
            variant="primary" 
            type="submit"
          >
            Accept
          </Button>
          <Button 
            variant="plain" 
            onClick={() => history.push("/attestmenu")}
          >
            Back
          </Button>
        </FormGroup>
      }
    </Form>
  )
};

export default CovidTestForm;
