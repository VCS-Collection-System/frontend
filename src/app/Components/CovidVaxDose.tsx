import React from "react";
import {
  Card,
  CardTitle,
  CardBody,
  DatePicker,
  FormGroup,
  FormSelect,
  FormSelectOption,
  GridItem,
  Text,
  TextInput,
  TextVariants
} from "@patternfly/react-core";
import { TimesIcon } from "@patternfly/react-icons";
import { Controller } from "react-hook-form";
import { 
  dateConvert, 
  americanDateFormat,
  covidVaxDateRules,
  covidVaxDateValidator
} from "@app/utils/utils";

const orderSuffix = ["st", "nd", "rd", "th"];
const minVaxDate = "06/01/2020";

const CovidVaxDose = (props) => {
  const doseText = props.doseNumber + orderSuffix[Math.min(props.doseNumber, orderSuffix.length)-1];

  return (
    <GridItem md={6} sm={12}>
      <Card>
        <CardTitle>{doseText} Dose or Booster {
          props.isRequired ? "" : <Text component={TextVariants.a} style={{float: "right"}} onClick={() => {props.removeDose(props.doseNumber)}} data-testid="remove-dose"><TimesIcon/></Text>
        }</CardTitle>
        <CardBody>
          <Controller
            name={"vax_date_" + props.doseNumber}
            control={props.control}
            defaultValue={props.isHRSubmit ? "" : (dateConvert(sessionStorage.getItem("vax_date_" + props.doseNumber) || ""))}
            render={({ field, fieldState }) => (
              <FormGroup
                helperText="The date you received the dose."
                helperTextInvalid="This field is required, please enter a valid date."
                fieldId={"shot-" + props.doseNumber + "-date"}
                label="Date"
                isRequired={props.isRequired}
                validated={fieldState.error ? "error" : "default"}
              >
                <DatePicker
                  aria-label={"shot-" + props.doseNumber + "-date"}
                  onChange={field.onChange}
                  dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
                  dateParse={americanDateFormat}
                  placeholder="MM-DD-YYYY"
                  value={field.value}
                  validators={
                    props.doseNumber == 1 ?
                      [ (date) => covidVaxDateValidator(date, minVaxDate, null) ] :
                      [ (date) => covidVaxDateValidator(date, minVaxDate, new Date(props.getValues("vax_date_" + (props.doseNumber-1)).replace(/-/g,'/'))) ]
                  }
                />
              </FormGroup>
            )}
            rules={{ required: props.isRequired, validate: (data) => covidVaxDateRules(data, minVaxDate, props.getValues("vax_date_" + (props.doseNumber-1))), pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/ }}
          />
          <Controller
            name={"vaccine_" + props.doseNumber}
            control={props.control}
            defaultValue={props.isHRSubmit ? "" : (sessionStorage.getItem("vaccine_" + props.doseNumber) || "")}
            render={({ field, fieldState }) => (
              <FormGroup
                helperText={
                  props.isHRSubmit ? 
                  "The vendor of this dose as shown on the documentation." : 
                  "The vendor of this dose."
                }
                helperTextInvalid="This field is required, please make a selection."
                isRequired={props.isRequired}
                label="Vaccine Manufacturer"
                fieldId={"vaccine-vendor-" + props.doseNumber}
                validated={fieldState.error ? "error" : "default"}
                style={{marginTop: "var(--pf-global--gutter--md)"}}
              >
                <FormSelect
                  aria-label={"vaccine-vendor-" + props.doseNumber}
                  value={field.value}
                  onChange={field.onChange}
                >
                  <FormSelectOption
                    isPlaceholder
                    isDisabled
                    label="Please Select"
                  />
                  {props.vendorOptions.map((vendorOption, index) => (
                    <FormSelectOption
                      key={index}
                      value={vendorOption.value}
                      label={vendorOption.label}
                    />
                  ))}
                </FormSelect>
              </FormGroup>
            )}
            rules={{ required: true }}
          />
          <Controller
            name={"lot_number_" + props.doseNumber}
            control={props.control}
            defaultValue={props.isHRSubmit ? "" : (sessionStorage.getItem("lot_number_" + props.doseNumber) || "")}
            render={({ field, fieldState }) => (
              <FormGroup 
                fieldId={"lot-number-" + props.doseNumber}
                helperText={
                  props.isHRSubmit ? 
                  "The lot/batch number of the vaccine received as shown on the documentation." : 
                  "The lot/batch number of the vaccine that you received."
                }
                helperTextInvalid="Please enter a valid lot/batch number."
                label="Lot/Batch Number"
                validated={fieldState.error ? "error" : "default"}
                style={{marginTop: "var(--pf-global--gutter--md)"}}
              >
                <TextInput
                  aria-label={"lot-number-" + props.doseNumber}
                  type="text"
                  value={field.value}
                  onChange={field.onChange}
                />
              </FormGroup>
            )}
            rules={{
              pattern: /^[A-Za-z0-9-]+$/
            }}
          />
        </CardBody>
      </Card>
    </GridItem>
  )
}

export default CovidVaxDose;