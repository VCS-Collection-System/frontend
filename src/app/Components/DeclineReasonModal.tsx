import * as React from 'react';
import { Button, Form, FormGroup, Modal, ModalVariant, Radio, Stack } from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';
import { Controller, useForm } from "react-hook-form";

const DeclineReasonModal = ({...props}) => {
  const { control, handleSubmit } = useForm();

  const declineReasons = [
    "Attachments are not legible",
    "Supporting documentation is missing required information",
    "Supporting documentation is not in an acceptable format",
    "Supporting documentation differs from entered employee or vaccination information",
    "Other (more information to be provided)"
  ]

  const onSubmit = (data) => {
    props.sendReview(false, data["decline-reason"]);
  }

  return (
    <Modal
      isOpen={props.isModalOpen}
      onClose={() => props.setIsModalOpen(false)}
      title="Reason For Decline - Select Most Appropriate"
      variant={ModalVariant.medium}
    >
      <Form onSubmit={handleSubmit(onSubmit)}>
        <Controller
          name="decline-reason"
          control={control}
          rules={{ required: true }}
          render={({ field, fieldState }) => (
            <FormGroup
              fieldId="decline-reason"
              isRequired
              validated={fieldState.error ? "error" : "default"}
              helperTextInvalid="* Please select a reason to proceed"
            >
              <Stack hasGutter>
                {declineReasons.map((reason, index) => (
                  <Radio
                    key={"radio-" + index}
                    name={"decline-reason-radio" + index}
                    id={"radio-controlled-" + index}
                    isChecked={field.value === reason}
                    onChange={(_, event) => field.onChange(event.currentTarget.value)}
                    label={reason}
                    value={reason}
                  />
                ))}
              </Stack>
            </FormGroup>
          )}
        />
        <FormGroup fieldId="decline-reason-buttons">
          <Button
            icon={<CheckCircleIcon />}
            isDisabled={props.isSendingReview}
            isLoading={props.isSendingReview}
            key="confirm"
            variant="primary"
            type="submit"
          >
            Confirm
          </Button>
          <Button
            key="cancel"
            onClick={() => props.setIsModalOpen(false)}
            variant="plain"
          >
            Cancel
          </Button>
        </FormGroup>
      </Form>
    </Modal>
  )
}

export default DeclineReasonModal;