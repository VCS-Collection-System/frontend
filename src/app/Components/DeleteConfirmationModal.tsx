import React from 'react';
import { Modal, Button } from '@patternfly/react-core';

const DeleteConfirmationModal = ({...props}) => {

 

    return (
      <React.Fragment>
        <Modal
          aria-label="My modal context"
          title="Delete Vaccine record"
          isOpen={props.isModalOpen}
          onClose={() => props.setIsModalOpen(false)}
           actions={[
            <Button key="confirm" variant="primary" onClick={() =>props.handleDelete()}>
              Confirm
            </Button>,
            <Button key="cancel" variant="link" onClick={() => props.setIsModalOpen(false)}>
              Cancel
            </Button>
          ]}
        >
          Do you want to delete this record?
        </Modal>
      </React.Fragment>
    );
  }


export default DeleteConfirmationModal;