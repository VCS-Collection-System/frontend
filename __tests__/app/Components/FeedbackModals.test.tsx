import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react'
import { GenericResponseModal, ResponseErrorModal }  from '../../../src/app/Components/FeedbackModals';

const mockSetIsModalOpen = jest.fn(); 

describe('<ResponseErrorModal />', ()=> {
    it('should be defined', () => {
        expect(ResponseErrorModal).toBeDefined()
    })

    it("should match existing snapshot", () => {
        expect(render(<ResponseErrorModal isModalOpen={true}/>).baseElement).toMatchSnapshot()
    })

    it("should render default error message", () => {
        render(
            <ResponseErrorModal 
                isModalOpen={true}
            />
        )

        expect(screen.queryByText("Server is unable to respond. Please try again later.")).toBeTruthy(); 
    })

    it("should render error 400 message", () => {
        render(
            <ResponseErrorModal 
                isModalOpen={true}
                errorState={400}
            />
        )

        expect(screen.queryByText("Form entry is incorrect. Please review form entries.")).toBeTruthy(); 
    })

    it("should be able to close modal with corner x", () => {
        render(
            <ResponseErrorModal 
                isModalOpen={true}
                setIsModalOpen={mockSetIsModalOpen}
            />
        )
        const closeButton = screen.getAllByRole('button', {name: /Close/i})[0]
        fireEvent.click(closeButton)
        expect(mockSetIsModalOpen).toBeCalledWith(false)
    })

    it("should be able to close modal with close button", () => {
        render(
            <ResponseErrorModal 
                isModalOpen={true}
                setIsModalOpen={mockSetIsModalOpen}
            />
        )
        const closeButton = screen.getAllByRole('button', {name: /Close/i})[1]
        fireEvent.click(closeButton)
        expect(mockSetIsModalOpen).toBeCalledWith(false)
    })
})

describe('<GenericResponseModal />', ()=> {
    it('should be defined', () => {
        expect(GenericResponseModal).toBeDefined()
    })

    it("should match existing snapshot", () => {
        expect(render(<GenericResponseModal isModalOpen={true}/>).baseElement).toMatchSnapshot()
    })

    it("should render messages", () => {
        render(
            <GenericResponseModal 
                isModalOpen={true}
                message={"MOCK MESSAGE"}
            />
        )

        expect(screen.queryByText("MOCK MESSAGE")).toBeTruthy(); 
    })

    it("should be able to close modal with corner x", () => {
        render(
            <GenericResponseModal 
                isModalOpen={true}
                setIsModalOpen={mockSetIsModalOpen}
            />
        )
        const closeButton = screen.getAllByRole('button', {name: /Close/i})[0]
        fireEvent.click(closeButton)
        expect(mockSetIsModalOpen).toBeCalledWith(false)
    })

    it("should be able to close modal with close button", () => {
        render(
            <GenericResponseModal 
                isModalOpen={true}
                setIsModalOpen={mockSetIsModalOpen}
            />
        )
        const closeButton = screen.getAllByRole('button', {name: /Close/i})[1]
        fireEvent.click(closeButton)
        expect(mockSetIsModalOpen).toBeCalledWith(false)
    })
})
