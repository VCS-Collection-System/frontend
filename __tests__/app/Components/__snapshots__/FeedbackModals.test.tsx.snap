// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericResponseModal /> should match existing snapshot 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  />
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-modal-part-7"
          aria-label="GenericResponseModal"
          aria-labelledby="pf-modal-part-5"
          aria-modal="true"
          class="pf-c-modal-box pf-m-sm"
          data-ouia-component-id="OUIA-Generated-Modal-small-6"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-5"
          role="dialog"
        >
          <button
            aria-disabled="false"
            aria-label="Close"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-6"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 352 512"
              width="1em"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          
          <div
            class="pf-c-modal-box__body"
            id="pf-modal-part-7"
          />
          <footer
            class="pf-c-modal-box__footer"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-tertiary"
              data-ouia-component-id="OUIA-Generated-Button-tertiary-6"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Close
            </button>
          </footer>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<ResponseErrorModal /> should match existing snapshot 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  />
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-modal-part-2"
          aria-label="ResponseErrorModal"
          aria-labelledby="pf-modal-part-0 pf-modal-part-1"
          aria-modal="true"
          class="pf-c-modal-box pf-m-sm"
          data-ouia-component-id="OUIA-Generated-Modal-small-1"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-0"
          role="dialog"
        >
          <button
            aria-disabled="false"
            aria-label="Close"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 352 512"
              width="1em"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <header
            class="pf-c-modal-box__header"
          >
            <h1
              class="pf-c-modal-box__title"
              id="pf-modal-part-1"
            >
              
              <span
                class="pf-c-modal-box__title-text"
              >
                Error Submitting Form
              </span>
            </h1>
          </header>
          <div
            class="pf-c-modal-box__body"
            id="pf-modal-part-2"
          >
            <span>
               Server is unable to respond. Please try again later. 
            </span>
          </div>
          <footer
            class="pf-c-modal-box__footer"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-tertiary"
              data-ouia-component-id="OUIA-Generated-Button-tertiary-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Close
            </button>
          </footer>
        </div>
      </div>
    </div>
  </div>
</body>
`;
